mutt_attach.c

```diff
@@ -681,15 +681,11 @@ int mutt_view_attachment(FILE *fp, struct Body *a, enum ViewAttachMode mode,
 
   if (!entry || !entry->xneomuttkeep)
   {
-    if (fp && !mutt_buffer_is_empty(tmpfile))
+    if ((fp && !mutt_buffer_is_empty(tmpfile)) || unlink_tempfile)
     {
       /* add temporary file to TempAttachmentsList to be deleted on timeout hook */
```

Comment:
Does this mean there could be a potential race condition between the background script and NeoMutt with sufficiently low `timeout` values?

Reply:
I believe so. If I insert a `sleep` or a `mutt_sleep` of 5 seconds (arbitrary value) before the unlinking line, the file displays normally.